src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +44,41 @@
      * Андрей - 600
      * София - 900</p>
      */
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.LL.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+        StringBuilder sb = new StringBuilder();
+        int[] array = new int[names.length];
+        LocalDate dateFromFormatted = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateToFormatted = LocalDate.parse(dateTo, formatter);
+
+        // dateFrom > dateTo
+        if (dateFromFormatted.isAfter(dateToFormatted)
+                || dateFromFormatted.isEqual(dateToFormatted)) {
+            throw new IllegalDateParametersException();
+        }
+        for (int i = 0; i < names.length; i++) {
+            for (int j = 0; j < data.length; j++) {
+                String[] splittedArray = data[j].split(" ");
+                if (names[i].equals(splittedArray[1])) {
+                    LocalDate localDate = LocalDate.parse(splittedArray[0],
+                            formatter);
+                    if ((localDate.isAfter(dateFromFormatted)
+                            || localDate.isEqual(dateFromFormatted))
+                            && (localDate.isBefore(dateToFormatted)
+                            || localDate.isEqual(dateToFormatted))) {
+                        array[i] += Integer.parseInt(splittedArray[2])
+                                * Integer.parseInt(splittedArray[3]);
+                    }
+                }
+            }
+            names[i] += " - " + array[i];
+        }
+        sb.append("Отчёт за период ").append(dateFrom).append(" - ").append(dateTo).append("\n");
+        for (String s : names) {
```

Comment:
try to create solution without this loop. you can append result in the previous loops